collections/duties.js

```diff
@@ -86,6 +86,46 @@ function deleteDutie(id) {
   });
 }
 
+function updateDutie(data, id) {
+  const resHandler = require("../start").server.getResHandler();
+  const collection = require("../start").dutiesCollection;
+
+  if (!utils.isMatchesRegex(/[0-9a-fA-F]{24}/, id)) {
+    resHandler.routeNotFound("the id that was sent is not legal, it needs to be of 24 chars in alpha-numeric");
+    return;
+  }
+
+  let leagalKeys = ["name", "location", "days", "constraints", "soldiersRequired", "value", "soldiers"];
+  for (let key in data) {
```

Comment:
there are better more efficient built in options of doing this check. a for loop in java script is like hitler jokes: cool and fun when your young. but when you grow up people expect you to outgrow them